deployment/cdk/opensearch-service-migration/lib/migration-assistance-stack.ts

```diff
@@ -41,6 +42,45 @@ export class MigrationAssistanceStack extends Stack {
         }
     }
 
+    validateAndReturnVPCSubnetsForMSK(vpc: IVpc, brokerNodeCount: number, azCount: number, specifiedSubnetIds?: string[]): string[] {
+        if (specifiedSubnetIds) {
+            if (specifiedSubnetIds.length !== 2 && specifiedSubnetIds.length !== 3) {
+                throw new Error(`MSK requires subnets for 2 or 3 AZs, but have detected ${specifiedSubnetIds.length} subnet ids provided with 'mskSubnetIds'`)
+            }
+            if (brokerNodeCount < 2 || brokerNodeCount % specifiedSubnetIds.length !== 0) {
+                throw new Error(`The MSK broker node count (${brokerNodeCount} nodes inferred) must be a multiple of the number of 
+                    AZs (${specifiedSubnetIds.length} AZs inferred from provided 'mskSubnetIds'). The node count can be set with the 'mskBrokerNodeCount' context option.`)
+            }
+            const selectSubnets = vpc.selectSubnets({
+                subnetFilters: [SubnetFilter.byIds(specifiedSubnetIds)]
+            })
+            return selectSubnets.subnetIds
+        }
+        if (azCount !== 2 && azCount !== 3) {
+            throw new Error(`MSK requires subnets for 2 or 3 AZs, but have detected an AZ count of ${azCount} has been provided with 'mskAZCount'`)
+        }
+        if (brokerNodeCount < 2 || brokerNodeCount % azCount !== 0) {
+            throw new Error(`The MSK broker node count (${brokerNodeCount} nodes inferred) must be a multiple of the number of 
+                AZs (${azCount} AZs inferred from provided 'mskAZCount'). The node count can be set with the 'mskBrokerNodeCount' context option.`)
+        }
+
+        let uniqueAzPrivateSubnets: string[] = []
+        if (vpc.privateSubnets.length > 0) {
+            uniqueAzPrivateSubnets = vpc.selectSubnets({
+                subnetType: SubnetType.PRIVATE_WITH_EGRESS,
+                onePerAz: true
+            }).subnetIds
+        }
+        let desiredSubnets
+        if (uniqueAzPrivateSubnets.length >= azCount) {
+            desiredSubnets = uniqueAzPrivateSubnets.slice(0, azCount).sort()
```

Comment:
I think we need to sort before slicing here

Reply:
Agreed, adjusted